### DISCUSSION TITLE
Controlling the number of decimal places.

### DISCUSSION THREAD

**Discussion by ryaaanmorgan** [nathanmarlor/foxess_modbus#251] [Ideas]
Is there a way to choose the number of decimal places the integral sensors have ? Typically I would round to 2 decimal places. 
I can deduce from some of the entity files that you pass an integer through for the rounding number but I don’t want to chance changing anything.
I realise this is a very stylistic idea, so I apologise in advance!

**Comment by calum-mcfarlane** [comment]
I think the latest versions of HA allow you to change the precision used to display each entity. So even if it's 5DP from the sensor, you can show it at 2DP.

**Comment by canton7** [comment]
![Animation](https://github.com/nathanmarlor/foxess_modbus/assets/568104/d5456ead-7684-4015-ba18-23ab4196f109)

**Comment by ryaaanmorgan** [comment]
On reflection, I feel I didn’t explain myself properly…

I meant that I’d prefer to see the totals increase incrementally by 0.01 kWh rather than by 0.1 kWh as it is currently.
I realise I can change how the figures are displayed in terms of decimal space through the UI, but doing this just adds extra zeros on the end.

I had kept the old Riemann integrals (written into the config file) from before using this integration but recently, something has changed/broken which has changed their behaviour. They have now started to reset and show as unavailable when the source sensor is zero. Of course being able to control those sensors via the config file, I could choose how granular I needed it to be.

Cheers lads!

**Reply by canton7** [reply]
Aah, you're referring to the fact that we set the `round` configuration to 2, when the default is 3.

I don't think there's any particular reason we did this. I'll revert it to 3.

**Reply by calum-mcfarlane** [reply]
I think I have had the same thing, now you describe it like that. For instance, 
![image](https://github.com/nathanmarlor/foxess_modbus/assets/115465775/0293c9a7-1d8e-4737-a45a-ae29adbf5d86)

I am still using the StealthChesnut integration purely because I've not found time to set aside to migrate to this one and resolve any potential issues that might come up. I mention it because I don't think it's related to this integration as such, might be a problem with HA since it definitely didn't used to do this.

Edit: Doh was trying to reply to @ryaaanmorgan

**Reply by ryaaanmorgan** [reply]
Ahh thanks Antony. Of course, it’s just personal preference so if it doesn’t work for the majority just ignore me! 😂

Yeah I tend to agree Calum. I’ve actually noticed the load total sensor on this integration has also done it where it’s reset.

![image](https://github.com/nathanmarlor/foxess_modbus/assets/55667216/c9ea8647-85e9-450b-9069-28ceb580a8fe)

**Reply by ryaaanmorgan** [reply]
> Aah, you're referring to the fact that we set the `round` configuration to 2, when the default is 3.
> 
> I don't think there's any particular reason we did this. I'll revert it to 3.

I saw you'd pushed the changes through, thanks Antony.

I've just looked over the Daily and Total sensors and unless I'm missing something, I'm not sure it's consistently been applied. 
The Load and PV sensors it has worked for the totals but not the daily sensors but for everything else, it's unchanged. 
Let me know if there is any specific testing you'd like me to do :)

**Reply by canton7** [reply]
Are you connected by a direct LAN cable or by RS485?

**Reply by ryaaanmorgan** [reply]
RS485 with a W610 device.

**Reply by canton7** [reply]
The only integration sensors we use there are for PV1 and PV2 (because the inverter doesn't supply totals) and Load Power (for legacy reasons - I want to change this). All of the other totals are read from the inverter

**Reply by ryaaanmorgan** [reply]
Ahh that makes sense, okay thanks Antony!

**Reply by canton7** [reply]
(and the inverter supplies them to the nearest 100Wh). I'd rather use totals from the inverter where possible because 1) they're more accurate, 2) they don't break if HA stops working for a bit, and 3) there's no risk of them randomly jumping around
